### PR TITLE
*: update tidb

### LINKED ISSUE #617: update tidb

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: close #617

### What is changed and how it works?
as title
ref https://github.com/pingcap/tidb/pull/33569
### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

File: pkg/dbutil/table_test.go [hunk 1]
```diff
@@ -169,5 +169,5 @@ func (*testDBSuite) TestSchemacmpEncode(c *C) {
 	c.Assert(err, IsNil)
 
 	table := schemacmp.Encode(tableInfo)
-	c.Assert(table.String(), Equals, "CREATE TABLE `tbl`(`id` INT(24) NOT NULL, PRIMARY KEY (`id`)) CHARSET UTF8MB4 COLLATE UTF8MB4_BIN")
+	c.Assert(table.String(), Equals, "CREATE TABLE `tbl`(`id` INT(24), PRIMARY KEY (`id`)) CHARSET UTF8MB4 COLLATE UTF8MB4_BIN")
```
Comment (Ehco1996) [inline pkg/dbutil/table_test.go]: i am not sure why in old test when create SQL is

```sql
CREATE TABLE `test`.`atest` (`id` int(24), primary key(`id`))
```
will return a 

```sql
CREATE TABLE `tbl`(`id` INT(24) NOT NULL, PRIMARY KEY (`id`)) CHARSET UTF8MB4 COLLATE UTF8MB4_BI
```

**Comment by Ehco1996** [comment]
ci passed ptal 🧡🧡

/cc @overvenus  @lance6716

**Comment by lance6716** [comment]
the purpose of this PR is to let release-5.X branch of DM update dependency?

**Review by lance6716 — APPROVED** [review]

**Comment by Ehco1996** [comment]
> the purpose of this PR is to let release-5.X branch of DM update dependency?

No, this pr will take effect on master in (6.x) release

**Comment by lance6716** [comment]
> > the purpose of this PR is to let release-5.X branch of DM update dependency?
> 
> No, this pr will take effect on master in (6.x) release

DM master branch uses the functions in pingcap/tidb package, not pingcap/tidb-tools

**Comment by Ehco1996** [comment]
> > > the purpose of this PR is to let release-5.X branch of DM update dependency?
> > 
> > 
> > No, this pr will take effect on master in (6.x) release
> 
> DM master branch uses the functions in pingcap/tidb package, not pingcap/tidb-tools

I wish that was the case, but unfortunately not 😭😭
https://github.com/pingcap/tiflow/blob/6b07e489842b4d011086ee112d04d471e64419c5/dm/chaos/cases/diff.go#L23

**Review by overvenus — APPROVED** [review]

**Pull request merged**